packages/components/tag/src/tag.tsx

```diff
@@ -65,60 +67,68 @@ export type TTagProps = {
    * Content rendered within the tag
    */
   children: ReactNode;
+  /**
+   * Indicates the color scheme of the tag.
+   */
+  tone: 'primary' | 'warning' | 'surface';
 };
 
 const defaultProps: Pick<
   TTagProps,
-  'type' | 'isDisabled' | 'isDraggable' | 'horizontalConstraint'
+  'tone' | 'isDisabled' | 'isDraggable' | 'horizontalConstraint'
 > = {
-  type: 'normal',
+  tone: 'primary',
   isDisabled: false,
   isDraggable: false,
   horizontalConstraint: 'scale',
 };
 
 const Tag = (props: TTagProps) => {
-  const linkProps =
-    props.to && !props.isDisabled ? { as: Link, to: props.to } : {};
+  let tagBodyProps;
+
+  switch (true) {
+    case props.isDisabled:
+      tagBodyProps = {};
+      break;
+    case Boolean(props.to):
+      tagBodyProps = { as: Link, to: props.to };
+      break;
+    case Boolean(props.onClick):
+      tagBodyProps = { as: 'button' as ElementType };
+      break;
+    default:
+      tagBodyProps = {};
+  }
+
```

Comment:
There should also be a warning if the user uses the deprecated `type` prop, similar to the [warning for size in PrimaryButton](https://github.com/commercetools/ui-kit/blob/c41a012c352e7ddb1d3a0ab4d4a9eb6af2d25583/packages/components/buttons/primary-button/src/primary-button.tsx#L120).

Reply:
Do we really need this? Implementation is easy, but honestly last time was a mess (since we broke peoples tests and forced them to act, although there was no breaking change).

I feel like once we have a plan how to remove deprecated stuff, we can still add this console.warns to all deprecated components/props and communicate a time- & deadline to teams. What do you think?

Reply:
that works for me